tempodb/encoding/compactor_block.go

```diff
@@ -121,6 +123,7 @@ func (c *CompactorBlock) BlockMeta() *backend.BlockMeta {
 
 	meta.StartTime = c.inMetas[0].StartTime
 	meta.EndTime = c.inMetas[0].EndTime
+	meta.FinalSize = c.totalBytesFlushed
```

Comment:
Why not use `.DataLength()` in the compactor block like the complete block?

Reply:
Good catch.  Was thinking the appendBuffer is flushed by the compactor block, but the appender is not.